x/swaprouter/router_test.go

```diff
@@ -941,3 +942,84 @@ func (suite *KeeperTestSuite) calcInAmountAsSeparateSwaps(osmoFeeReduced bool, r
 		return nextTokenIn
 	}
 }
+
+func (suite *KeeperTestSuite) TestSingleSwapExactAmountIn() {
+	tests := []struct {
+		name                   string
+		poolId                 uint64
+		poolCoins              sdk.Coins
+		poolFee                sdk.Dec
+		tokenIn                sdk.Coin
+		tokenOutDenom          string
+		tokenOutMinAmount      sdk.Int
+		expectedTokenOutAmount sdk.Int
+		expectError            bool
+	}{
+		{
+			name:                   "One route: Swap - [foo -> bar], 1 percent fee",
+			poolId:                 1,
+			poolCoins:              sdk.NewCoins(sdk.NewCoin(foo, defaultInitPoolAmount), sdk.NewCoin(bar, defaultInitPoolAmount)),
+			poolFee:                defaultPoolSwapFee,
+			tokenIn:                sdk.NewCoin(foo, sdk.NewInt(100000)),
+			tokenOutMinAmount:      sdk.NewInt(1),
+			tokenOutDenom:          bar,
+			expectedTokenOutAmount: sdk.NewInt(98999),
```

Comment:
Can you please add a comment in terms of how this expected value is calculated? If possible, a WolframAlpha link with calculations would be helpful